src/odemis/odemisd/main.py

```diff
@@ -328,71 +329,98 @@ def _terminate_all_alive(self):
         It also stops the containers, once no component is running in them.
         """
         mic = self._instantiator.microscope
-        alive = set(mic.alive.value)
-
-        # create a "graph" child->parents
-        parents = {c: set() for c in alive}  # comp -> set of comps (all its parents)
-        for comp in alive:
-            # TODO: if a component was created by delegation, use its creator
-            # instead of itself as "used" by the other components, to ensure
-            # they might not be terminated too early.
+
+        # create a "graph" dependency->parents
```

Comment:
The "parent" term only matches for the children. Here we are talking of dependencies. So it should be something like "dependents".  So I'd suggest to rename the `_parents` variable as `_dependents`.
Moreover, this is not a graph (my fault). So I'd update the comment as something like:
`# alive component -> components which depends on it`